### ISSUE TITLE
4 tests reporting incorrect test count information

### ISSUE THREAD

**Issue by jkeenan** [atoomic/perl#249]
As of commit 138c2a89e0 in the `alpha-dev-02-strict` branch, several test
files appear to be passing all individual tests but evidencing problems with
the count of tests.
```
t/io/pipe ...................................................... 
 # Looks like you planned 27 tests but ran 15.
ok
```
```
t/op/getppid ................................................... 
 # Looks like you planned 8 tests but ran 0.
 # Looks like you planned 8 tests but ran 0.
 # Looks like you planned 8 tests but ran 3.
 # Looks like you planned 8 tests but ran 3.
ok
```
```
t/op/magic .....................................................
 # Looks like you planned 192 tests but ran 87.
ok
```
```
t/op/write ..................................................... 
 # Looks like you planned 624 tests but ran 590.
ok
```
These need further investigation.

**Comment by atoomic** [comment]
a git bisect shows that this is broken by 2082e415c44f35f7c37f6df51c3bfccc4861fa82
@jkeenan this is coming from the changes from 2082e415c44f35f7c37f6df51c3bfccc4861fa82

**Comment by atoomic** [comment]
fixed by 5c0671ab

**Comment by jkeenan** [comment]
> a git bisect shows that this is broken by [2082e41](https://github.com/atoomic/perl/commit/2082e415c44f35f7c37f6df51c3bfccc4861fa82)
> @jkeenan this is coming from the changes from [2082e41](https://github.com/atoomic/perl/commit/2082e415c44f35f7c37f6df51c3bfccc4861fa82)

Did you use `git bisect` directly for this?  (I haven't been able to get `Porting/bisect.pl` to work yet.)  If so, could you post your invocation?

> fixed by [5c0671a](https://github.com/atoomic/perl/commit/5c0671ab2c0201b779371934bd30e7e0f7d43512)

Yes, after I got afk I realized the answer had to lie in `t/test.pl` but by then I was too sleepy to investigate it further.

I have confirmed that the fix works.

Thank you very much.
Jim Keenan

**Comment by atoomic** [comment]
Hi @jkeenan I indeed used `git bisect` by itself, here are the basic instructions to use it

```
> git bisect start broken-git-id working-git-id
# then run your test, in my case, I avoided compiling Perl as I suspected it was a unit test only issue
# if it succeeds
> git bisect good
# if it fails
> git bisect bad
# if current state cannot be tested use skip
> git bisect skip        

# repeat: run test, tag as good or bad...
# until you reach the offending commit
# you can also use 
> git bisect visualize
> git bisect log

# to exit the bisect process
> git bisect reset
```
